packages/terra-date-input/src/DateInput.jsx

```diff
@@ -442,6 +468,32 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Sets state to new value dependent on existing date
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {Number} addDays Adds days to current date or today's date.
+   */
+  setHotKeyDate(event, addDays) {
+    const validYear = (this.state.year.length === 4 && Number(this.state.year) < DateInputUtil.MaxYearValue && Number(this.state.year) > DateInputUtil.MinYearValue);
+    const validDay = (Number(this.state.day) <= 31 && Number(this.state.day) > 0);
```

Comment:
The date-input as it is today allows to use the Up arrow key to increase the day to 31 regardless how many days are in a given month. For example, February has only 28 days on non-leap years but it allows to enter up to 31.  When this occur and the user presses the `+` key, instead of updating the date `March 01 2021` it would actually update it to `March 04 2021`. So it is actually 3 days (31 - 28) ahead of what it should be. We should discuss how to best handle this type of scenario.


In this workflow, I pressed the UP key several times to increase the day to 31 and then press the `+` just once. It advanced the date to `March, 04 2021`

![date-input-hotkeys](https://user-images.githubusercontent.com/22618655/141326980-2cf1c5be-b9e2-4032-841b-602c76185bca.gif)

Reply:
The changes here https://github.com/cerner/terra-framework/pull/1483/commits/64ce6f898cee6c84c6a041722c74baa14a2d84c7#diff-ea35ec190c52db171ef5f774b071e80db0b038abff0e19580b310959fc582329R479 should deal with this. We now consider a date that has more days in it then the maximum an "invalid" date.

Reply:
I think making days that exceed the total number of days in the month as invalid makes sense. However, from an end user's perspective the new change doesn't seem like what they would expect.

Using the same workflow as above, instead of advancing the date to `March, 04 2021` like what it did before it now changes the date from the invalid `February, 31 2021` date to tomorrow's date (December, 09, 2021) when pressing the `+` key. I think the user would expect it to advance the date to `March, 01 2021`.

This is what I'm thinking but would like to get @neilpfeiffer and @chrismichalewicz 's thoughts.
- When we have an incomplete date (missing either a month, day, or year), pressing `-`/`+` should change to yesterday/tomorrow respectively.

- When we have an invalid date (have month, day and year but the date is not a valid date; e.g,; outside the range of a valid date), pressing `-`/`+` should change to the previous/next valid date respectively.

If the current behavior is not what we want, we could check if we have a 2 digit day that exceeds the total number of days in a month and if so advance to the first day of the next month.

![date_input](https://user-images.githubusercontent.com/22618655/145236444-88df8fd9-57b6-422d-8ad4-308763c8574b.gif)

Reply:
We discussed this scenario with the team and decided that the `-`/`+` keys will always change to yesterday/tomorrow respectively regardless if the value is incomplete or invalid.

Reply:
Doing any kind of partial interpretation may become extremely difficult and challenging with any additional key shortcut keys we add (start/end of month, start/end of week, stand/end of year, etc), trying to make assumptions for what a user intended may create some very complex log and never be perfectly correct - therefore it will be the simplest and most consistent to function with _*incomplete*_ and _*invalid*_ as the same response and ignore any entered information and use today as the reference value. 

Only a complete and valid/accurate date should result in using that date as the reference value instead of today/now.